### ISSUE TITLE
Write tutorial

### ISSUE THREAD

**Issue by peterfriese** [applause/applause#11]

**Comment by mkTheSICK** [comment]
hi peter & heiko,

i am trying to generate my first applause project through the existing sample file. i have set up my eclipse just like written in the short installation guide. in order to generate android-code i am missing the specific mwe2-workflow file in the applause editor like i am used to from other xtext projects. would you please write a short getting started or simple textual tutorial to help me out on this one? i am currently working on my thesis where i want to use applause to extend the functionality of an existing gui-editor to automatically generate mobile apps (mainly android).

thanks for your work &
kind regards,
martin

**Comment by peterfriese** [comment]
Martin,

thanks for using Applause!

I understand you're looking for MWE workflow files to launch the code generator. As you might have noticed, we're making use of the incremental project builder infrastructure to allow for a seamless user experience. In other words, the code generators will be run automatically as soon as you save a file.

The way it works is like this:
- The master builder walks all projects in the workspace.
  - If a project has a project dependency to the model project, it will be considered a generator target / platform project
    - The builder will then try to find a matching platform generator by asking all of the registered generators if they can generate code for the current project

HTH,
Peter

PS: we _will_ write a tutorial, promised :-)

**Comment by mluebeck** [comment]
Hi,

I want to use Applause for a company project, and habe finished step 1-4 of the installation
1. Eclipse + Xtext 1.0.2 installed
2. Clone APPlause done
3. Start Eclipse
4. Import all projects in the tooling subdirectory into your workspace.

After that, i got many red spots in eclipse (aka errors ;-) )
Is there a version issue? Or must i fix it in the build settings? 

I can send you a pic, if it helps.

After that is fixed, I can support you in writing some documentation ;-)

kind regards

Mario

**Comment by alopix** [comment]
I have the same problems. Many many errors :(

**Comment by peterfriese** [comment]
Did you use Eclipse + Xtext 1.0.2 (you can get a a copy here: http://download.itemis.com/distros/

Anyway, I am currently busy ramping up for the next version of Applause, you should see the first results in the coming weeks.

**Comment by alopix** [comment]
Yes, I downloaded the latest version but it wouldn't work.
org.applause.lang.ui generates a lot of "_fsp cannot be resolved to a variable (InternalApplauseDslParser.java) and then there are a lot of other problems in org.applause.land and.ui
Don't know if it helps. Could be that the problems only occur in Eclipse Version: 3.7.1 (your download links are for 3.6)?!

I'm currently evaluating APPlause as part of my Master thesis and as far as I've seen, it's great, but not being able to test is kind of a bummer ;)
Wish you all the best for the upcoming version - I'm definitely going to check it out.

**Comment by peterfriese** [comment]
You need to make sure you use Xtext 1.0.2 as specified in the install instructions. Xtext has changed a lot in the past which is why you get so many compile errors.

**Comment by greenmobile** [comment]
hey peter,

yesterday i met Dr. Georg Pietrek in one of the university lectures (model driven development) I hear in Marburg (Germany), where he gave us a short introduction into development at itemis. I think mdd is very interestant especially for app-development (I am not sure at all, if the complexity of program logic can be set in a DSL...I would like to hear your opinion). So I started searching and did find applause today.

I am a founder of a small company in brunswick that develops mobile apps for ecological sustainable mobility. I am very interested in what you have developed and would like to hear more about it. We tried similiar things at our company and builded our own persistence framework that generates Java Code for Android and GWT. Maybe we can share our experiences... 

There is one question left: Are there any news about the tutorial?

best regards,
Tillmann Heigel
